### PR TITLE
Fix "line/bar chart" tooltip flake

### LINKED ISSUE #43936: [Flaky Test]: scenarios > visualizations > line/bar chart > tooltips > single series question grouped by month on dashboard should not show percent change when previous month is missing from result data
Last Flake: https://github.com/metabase/metabase/actions/runs/9437892268
Last Flake Time: 2024-06-09T09:51:00-07:00
Flakes in the last day: 1
Flakes in the last 3d: 1
Flakes in the last 7d: 1

### PR DESCRIPTION
**Pull request by nemanjaglumac**
Resolves #43936

Stress test: https://github.com/metabase/metabase/actions/runs/9466063592

### CODE REVIEW

File: e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js [hunk 1]
```diff
@@ -762,15 +729,21 @@ function setupDashboard(cardId, addedSeriesCardId) {
   });
 }
 
-function showTooltipForCircleInSeries(seriesColor, index = 0) {
-  cartesianChartCircleWithColor(seriesColor).eq(index).realHover();
+function resetHoverState() {
+  echartsTriggerBlur();
+  cy.get(POPOVER_ELEMENT).should("not.exist");
+  cy.wait(50);
```
Comment (alxnddr) [inline e2e/test/scenarios/visualizations-charts/line-bar-tooltips.cy.spec.js:735]: Without this line it has become way better but still flaked 1 out of 5 runs sometimes

**Review by alxnddr — APPROVED** [review]

**Comment by alxnddr** [comment]
@nemanjaglumac reviewed my changed

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts [hunk 2]
```diff
@@ -114,6 +115,8 @@ export function useModelsAndOption({
       return {};
     }
 
+    const shouldAnimate = !isPlaceholder && !isCypressActive;
+
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts:119]: Coping my comment from #43984: any chance we could use the `@prefers-reduced-motion` media setting for this? IIRC it's already disabled in Cypress, and I suppose people won't expect charts to be animated when the setting is on as well

**Review by kulyk — APPROVED** [review]

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts (hunk 2 above)
Comment (alxnddr) [reply frontend/src/metabase/visualizations/visualizations/CartesianChart/use-models-and-option.ts:119]: @kulyk great idea, I'll explore it and add to https://github.com/metabase/metabase/pull/43984

**Pull request merged**